src/HotChocolate/AspNetCore/src/Transport.Http/Helper/HttpMessageHelper.cs

```diff
@@ -0,0 +1,71 @@
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using HotChocolate.Transport.Abstractions;
+using HotChocolate.Transport.Abstractions.Helpers;
+
+namespace HotChocolate.Transport.Http.Helper;
+
+internal static class HttpMessageHelper
+{
+    public static HttpRequestMessage AddDefaultAcceptHeaders(this HttpRequestMessage requestMessage)
+    {
+        requestMessage.Headers.Add("Accept", "application/graphql-response+json; charset=utf-8");
+        requestMessage.Headers.Add("Accept", "application/json; charset=utf-8");
+        return requestMessage;
+    }
+
+    public static HttpRequestMessage AddJsonBody(
+        this HttpRequestMessage requestMessage,
+        OperationRequest request)
+    {
+        using var memoryStream = new MemoryStream();
+        using var jsonWriter = new Utf8JsonWriter(memoryStream, JsonDefaults.WriterOptions);
+        jsonWriter.WriteStartObject();
+
+        if(request.Id is not null)
```

Comment:
As a start I copied these implementation for serilization of the MessageHelper class in the websocket implementation. Does it makes sense to move it to the abstractions? 

As far as I understand it, we need to do it manually and cannot use the JsonSerializer.Serialize methode because we need to skip properties on OeprationRequest which are null. Is that right?

Reply:
We also do not use the memory stream as it allocates ... we have an internal BufferWriter that rents memory.

Reply:
I updated it by using the ArrayWriter from HotChocolate.Utilities which is used by Fusion too